lib/rules/require-key-for-conditionally-rendered-repeated-components.js

```diff
@@ -0,0 +1,371 @@
+  if (node.parent && node.parent.type === 'VElement') {
+    const conditionalFamily = conditionalFamilies.get(node.parent)
+
+    if (
+      conditionalFamily &&
+      (utils.hasDirective(node, 'bind', 'key') ||
+        utils.hasAttribute(node, 'key') ||
+        !hasConditionalDirective(node) ||
+        !isConditionalFamilyComplete(conditionalFamily))
+    ) {
+      return
+    }
+
+    context.report({
+      node: node.startTag,
+      loc: node.startTag.loc,
+      messageId: 'requireKey',
+      data: {
+        componentName
+      },
+      fix(fixer) {
+        const afterComponentNamePosition =
+          node.startTag.range[0] + componentName.length + 1
+        return fixer.insertTextBeforeRange(
+          [afterComponentNamePosition, afterComponentNamePosition],
+          ` key='${uniqueKey}'`
+        )
+      }
+    })
+  }
+}
+
+/**
+ * Checks for the presence of conditional directives such as 'v-if', 'v-else-if', or 'v-else' in the
+ * given Vue component node.
+ *
+ * @param {VElement} node - The node to check for conditional directives.
+ * @returns {boolean} Returns true if a conditional directive is found in the node or its parents,
+ *   false otherwise.
+ */
+const hasConditionalDirective = (node) =>
+  utils.hasDirective(node, 'if') ||
+  utils.hasDirective(node, 'else-if') ||
+  utils.hasDirective(node, 'else')
+
+/**
+ * Checks whether a conditional family (comprising of 'if', 'else-if', and 'else' conditions) is complete.
+ * A conditional family is considered complete if:
+ * 1. An 'if' condition is present.
+ * 2. All elements of the family (if, else, else-if) share the same rawName.
+ * 3. There exists either an 'else' condition or at least one 'else-if' condition.
+ *
+ * @param {{ if: VElement | null, elseIf: VElement[], else: VElement | null }} family - The conditional family object with 'if', 'else-if', and 'else' properties.
+ * @returns {boolean} True if the conditional family is complete based on the above criteria, false otherwise.
+ */
+const isConditionalFamilyComplete = (family) => {
+  if (!family || !family.if) {
+    return false
+  }
+  const familyComponentName = family.if.rawName
+  if (family.else && family.else.rawName !== familyComponentName) {
+    return false
+  }
+  if (family.elseIf.some((node) => node.rawName !== familyComponentName)) {
+    return false
+  }
+  return Boolean(family.else || family.elseIf.length > 0)
+}
+
+// =============================================================================
+// Rule Definition
+// =============================================================================
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'require key attribute for conditionally rendered repeated components',
+      categories: null,
+      recommended: false,
+      url: 'https://eslint.vuejs.org/rules/require-key-for-conditionally-rendered-repeated-components.html'
```

Comment:
The rule name is *really* long, even when compared to our longest rule name `vue/no-spaces-around-equal-signs-in-attribute`. Why not `v-if-else-key` like suggested in the issue?

Reply:
ahh, naming can be difficult and sometimes I try too hard, I will change it to the suggested name

Reply:
[c4871c6](https://github.com/vuejs/eslint-plugin-vue/pull/2280/commits/c4871c603e145253ffb00b4b2f7ecc887fa99da1)